services/dynamic-help-service.ts

```diff
@@ -0,0 +1,26 @@
+///<reference path="../../.d.ts"/>
+"use strict";
+
+import os = require("os");
+
+export class DynamicHelpService implements IDynamicHelpService {
+	constructor(private $injector: IInjector) { }
+	public isProjectType(...args: string[]): IFuture<boolean> {
+		return ((): boolean => {
+			// resolve here as Project.IProject is not part of common-lib
+			var $project: Project.IProject = this.$injector.resolve("project");
```

Comment:
My suggestion is to add `dynamicHelpProvider` as a private filed of `dynamicHelpService`.
```Typescript
public class DynamicHelpService {
     constructor(private $dynamicHelpProvider: IDynamicHelpProvider) { }
     
    public isProjectType(...args: string[]): IFuture<boolean> {
        return this.$dynamicHelpProvider.isProjectType(args);
    }
}
```
The implementation of `dynamicHelpProvider` in tns would be:
```Typescript
public class DynamicHelpProvider {
   public isProjectType(...args: string[]): IFuture<boolean> {
       return Future.fromResult(true);
   }
}
```
And the implementation in icenium-cli would be:

```Typescript
public class DynamicHelpProvider {
    constructor(private $project: Project.IProject) { }

   public isProjectType(...args: string[]): IFuture<boolean> {
         var framework = $project.projectData.Framework.toLowerCase();
         return _.any(args, arg => arg.toLowerCase() === framework);
   }
}
```